File: icom/icom_client.py
#!/usr/bin/python

import hashlib
import ssl
import websocket
import time
import _thread
import json

app_code = '123456789'
app_id = str(int(time.time()))

ws = websocket.WebSocket(sslopt = { "cert_reqs": ssl.CERT_NONE })
ws.connect(
    'wss://localhost:6030/app/'
    + hashlib.md5(str.encode(str(int(time.time())) + ':' + app_code)).hexdigest()
)

def receive_messages(websock):
    while True:
        try:
            message = websock.recv()
            print(">", message)
            message = json.loads(message)
            if(message['ping'] and message['ping'] != app_id):
                websock.send(json.dumps({'pong': app_id}))
            elif(message['recv']):
                if(message['send']):
                    if(message['type'] and message['data']):
                        print(message['data'])
                    elif(message['send'] == app_id):
                            websock.send(json.dumps({
                                'send': message['recv'], 'recv': app_id,
                                'type': 'text/plain',
                                'data': {'id': 0, 'name': 'user'}
                            }))
        except Exception as exception:
                print(exception)
                #break

_thread.start_new_thread(receive_messages, (ws, ))  # прослушивание с сервера
ws.send(json.dumps({'ping': app_id}))
while True:
    to_send = input()
    ws.send(to_send)

File: icom/icom_client.js
import path from 'path';
import { fileURLToPath } from 'url';
import WebSocket from 'ws';
import crypto from 'crypto';
import readline from 'node:readline';
import { stdin, stdout } from 'node:process';

const app_code = '123456789';
let app_id = new String(Math.floor(new Date().getTime() / 1000));

let app_hash = crypto.createHash('md5').update(
    new String(Math.floor(new Date().getTime() / 1000)) + ':' + app_code
).digest('hex');
const ws = new WebSocket(`wss://127.0.0.1:6030/app/${app_hash}` , {
    rejectUnauthorized: false
});

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

console.log(`App ID: ${app_id}`);

ws.on('error', console.error);

ws.on('message', function message(data) {
    console.log(`Data: ${data}`);
    let json = JSON.parse(data);
    if(json.ping) {
        ws.send(`{"pong": "${app_id}"}`);
    }
    else if(json.type && json.data) {
        if(json.send && json.recv && json.send == app_id) {
            console.log("Match");
        }
        else if(json.recv) {
        }
    }
    else if(!json.type && !json.data) {
        if(json.send && json.recv && json.send == app_id) {
            ws.send(JSON.stringify({
                "send": app_id, "recv": json.send,
                "type": "text/html", "data": "<div>Test</div>"
            }));
        }
    }
});

ws.on('open', function open() {
    ws.send(`{"ping": "${app_id}"}`);
});

const read = readline.createInterface({ input: stdin, output: stdout });

console.log('>');
read.on('line', (input) => {
    ws.send(input);
    console.log('>');
});
